addons/mail/models/mail_template.py

```diff
@@ -623,7 +624,10 @@ def send_mail(self, res_id, force_send=False, raise_exception=False, email_value
             attachment_ids.append((4, Attachment.create(attachment_data).id))
         if attachment_ids:
             mail.write({'attachment_ids': attachment_ids})
-
-        if force_send:
-            mail.send(raise_exception=raise_exception)
+        try:
+            if force_send:
+                mail.send(raise_exception=raise_exception)
+        except MailDeliveryException as mde:
+            if len(mde.args) == 2 and isinstance(mde.args[1], ConnectionError):
+                raise UserError(_("Check outgoing mail server, is it successfully configured or not."))
```

Comment:
My suggestion was based on the message. Basically if the message is specific, the exception should be specific too.
We should always make sure we raise something here since it would be confusing otherwise.
```suggestion
            if len(mde.args) == 2 and isinstance(mde.args[1], ConnectionRefusedError):
                 raise UserError(_("Could not contact the mail server, please check your outgoing email server configuration"))
            else:
                 raise UserError(_("There was an error when trying to deliver your Email, please check your configuration"))
```

Reply:
Hello @thiryr 
changes have been done, kindly review them.
Thanks.

Reply:
Hey @sauc-odoo please have a look at the review message as the one thing I want to change is described there :)
(or respond if the change isn't ok for you, or not clear)